src/kv/RocksDBStore.cc

```diff
@@ -619,56 +780,83 @@ void RocksDBStore::RocksDBTransactionImpl::set(
   const char *k, size_t keylen,
   const bufferlist &to_set_bl)
 {
+  bool is_default_cf;
+  rocksdb::ColumnFamilyHandle* cf_handle =
+    db->get_column_family_handle(prefix, is_default_cf);
+
   string key;
   combine_strings(prefix, k, keylen, &key);
 
   // bufferlist::c_str() is non-constant, so we can't call c_str()
   if (to_set_bl.is_contiguous() && to_set_bl.length() > 0) {
-    bat.Put(rocksdb::Slice(key),
+    bat.Put(cf_handle, rocksdb::Slice(is_default_cf ? key : string(k, keylen)),
 	     rocksdb::Slice(to_set_bl.buffers().front().c_str(),
 			    to_set_bl.length()));
   } else {
-    rocksdb::Slice key_slice(key);
+    rocksdb::Slice key_slice(is_default_cf ? key : string(k, keylen));
     rocksdb::Slice value_slices[to_set_bl.buffers().size()];
-    bat.Put(nullptr, rocksdb::SliceParts(&key_slice, 1),
+    bat.Put(cf_handle, rocksdb::SliceParts(&key_slice, 1),
             prepare_sliceparts(to_set_bl, value_slices));
   }
 }
 
 void RocksDBStore::RocksDBTransactionImpl::rmkey(const string &prefix,
 					         const string &k)
 {
-  bat.Delete(combine_strings(prefix, k));
+  bool is_default_cf;
+  rocksdb::ColumnFamilyHandle* cf_handle =
+    db->get_column_family_handle(prefix, is_default_cf);
+
+  string key = combine_strings(prefix, k);
+  bat.Delete(
+    cf_handle, rocksdb::Slice(is_default_cf ? key : k));
 }
 
 void RocksDBStore::RocksDBTransactionImpl::rmkey(const string &prefix,
 					         const char *k,
 						 size_t keylen)
 {
+  bool is_default_cf;
+  rocksdb::ColumnFamilyHandle* cf_handle = 
+    db->get_column_family_handle(prefix, is_default_cf);
+
   string key;
   combine_strings(prefix, k, keylen, &key);
-  bat.Delete(key);
+  bat.Delete(
+    cf_handle, rocksdb::Slice(is_default_cf ? key : string(k, keylen)));
 }
 
 void RocksDBStore::RocksDBTransactionImpl::rm_single_key(const string &prefix,
 					                 const string &k)
 {
-  bat.SingleDelete(combine_strings(prefix, k));
+  bool is_default_cf;
+  rocksdb::ColumnFamilyHandle* cf_handle =
+    db->get_column_family_handle(prefix, is_default_cf);
```

Comment:
assert for cf_handle? applicable to all references.

Reply:
ok